liquibase-standard/src/main/java/liquibase/changelog/visitor/UpgradeCheckSumVisitor.java

```diff
@@ -0,0 +1,57 @@
+package liquibase.changelog.visitor;
+
+import liquibase.ChecksumVersions;
+import liquibase.Scope;
+import liquibase.change.CheckSum;
+import liquibase.changelog.ChangeLogHistoryService;
+import liquibase.changelog.ChangeLogHistoryServiceFactory;
+import liquibase.changelog.ChangeSet;
+import liquibase.changelog.DatabaseChangeLog;
+import liquibase.changelog.filter.ChangeSetFilterResult;
+import liquibase.database.Database;
+import liquibase.exception.LiquibaseException;
+import liquibase.executor.Executor;
+import liquibase.executor.ExecutorService;
+import liquibase.executor.LoggingExecutor;
+import liquibase.statement.core.UpdateChangeSetChecksumStatement;
+
+import java.util.Set;
+
+public class UpgradeCheckSumVisitor implements ChangeSetVisitor {
+
+    private final Database database;
+
+    public UpgradeCheckSumVisitor(Database database) {
+        this.database = database;
+    }
+
+    @Override
+    public Direction getDirection() {
+        return Direction.FORWARD;
+    }
+
+    @Override
+    public void visit(ChangeSet changeSet, DatabaseChangeLog databaseChangeLog, Database database,
+                      Set<ChangeSetFilterResult> filterResults) throws LiquibaseException {
+        Executor executor = Scope.getCurrentScope().getSingleton(ExecutorService.class).getExecutor("jdbc", database);
+
+        ChangeLogHistoryService changeLogService = ChangeLogHistoryServiceFactory.getInstance().getChangeLogService(database);
+        if (changeLogService.isDatabaseChecksumsCompatible() || changeSet.getStoredCheckSum() == null ||
+                changeSet.getStoredCheckSum().getVersion() == ChecksumVersions.latest().getVersion()) {
+            return;
+        }
```

Comment:
Why do we return if the stored checksum is null?

Reply:
Hm, right. If it's null we should upgrade it, makes sense!